src/na/na_cci.c

```diff
@@ -1919,6 +1923,13 @@ na_cci_progress(na_class_t * na_class, na_context_t * context,
 		hg_time_t	t1, t2;
 		cci_event_t	*event = NULL;
 
+		struct pollfd pfd;
+
+		pfd.fd = NA_CCI_PRIVATE_DATA(na_class)->fd;
+		pfd.events = POLLIN;
+
+		poll(&pfd, 1, ((int) remaining) * 1000);
+
 		hg_time_get_current(&t1);
```

Comment:
This hg_time_get_current(&t1) call needs to be moved to *before* the poll call, so that we correctly measure how much time the poll() consumed.